cmd-script.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Python script binding for function entry and exit
+ *
+ * Copyright (C) 2017, LG Electronics, Honggyu Kim <hong.gyu.kim@lge.com>
+ *
+ * Released under the GPL v2.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdio_ext.h>
+
+#include "uftrace.h"
+#include "utils/utils.h"
+#include "utils/symbol.h"
+#include "utils/filter.h"
+#include "utils/fstack.h"
+#include "utils/script.h"
+
+#include "libtraceevent/event-parse.h"
+
+
+static int run_script_for_rstack(struct ftrace_file_handle *handle,
+				 struct ftrace_task_handle *task,
+				 struct opts *opts)
+{
+	struct uftrace_record *rstack = task->rstack;
+	struct uftrace_session_link *sessions = &handle->sessions;
+	struct sym *sym = NULL;
+	char *symname = NULL;
+
+	if (task == NULL)
+		return 0;
+
+	sym = task_find_sym(sessions, task, rstack);
+	symname = symbol_getname(sym, rstack->addr);
+
+	if (opts->kernel_skip_out) {
+		/* skip kernel functions outside user functions */
+		if (!task->user_stack_count && is_kernel_record(task, rstack))
+			goto out;
+	}
+
+	task->timestamp_last = task->timestamp;
+	task->timestamp = rstack->time;
+
+	if (rstack->type == UFTRACE_ENTRY) {
+		struct ftrace_task_handle *next = NULL;
+		struct fstack *fstack;
+		int rstack_depth = rstack->depth;
+		int depth;
+		struct ftrace_trigger tr = {
+			.flags = 0,
+		};
+		int ret;
+
+		ret = fstack_entry(task, rstack, &tr);
+		if (ret < 0)
+			goto out;
+
+		/* display depth is set in fstack_entry() */
+		depth = task->display_depth;
+
+		fstack = &task->func_stack[task->stack_count - 1];
+
+		if (task == next &&
+		    next->rstack->depth == rstack_depth &&
+		    next->rstack->type == UFTRACE_EXIT) {
+
+			/* leaf function - also consume return record */
+			fstack_consume(handle, next);
+
+			/* fstack_update() is not needed here */
+			fstack_exit(task);
+		}
```

Comment:
It seems this block is unnecessary.  The reason doing this in replay is to skip filtered entries and to find matching exit record in order to show a leaf function on a single line.  But `uftrace script` doesn't need to do it and simply to call entry and exit callbacks everytime.

As you can see the `next` is never set and finding symbol name below is redundant.